static/js/controllers/reports.js

```diff
@@ -92,36 +116,68 @@ var _ = require('underscore'),
         return results;
       };
 
-      var updateDisplayFields = function(report) {
+      var getDisplayFields = function(report) {
         // calculate fields to display
+        if (!report.fields) {
+          return [];
+        }
         var label = 'report.' + report.form;
         var fields = getFields([], report.fields, label, 0);
         var hide = report.hidden_fields || [];
         hide.push('inputs');
-        fields = _.reject(fields, function(field) {
+        return _.reject(fields, function(field) {
           return _.some(hide, function(h) {
             return field.label.indexOf(label + '.' + h) === 0;
           });
         });
-        $scope.displayFields = fields;
       };
 
-      $scope.setSelected = function(doc) {
-        if (doc.fields) {
-          updateDisplayFields(doc);
+      var setActionBar = function() {
+        var model = {};
+        model.selected = $scope.selected.map(function(s) {
+          return s.report || s.summary;
+        });
+        if (!$scope.selectMode &&
+            model.selected &&
+            model.selected.length === 1) {
+          var doc = model.selected[0];
+          model.verified = doc.verified;
+          model.type = doc.content_type;
+          model.sendTo = doc;
         }
+        $scope.setActionBar(model);
+      };
 
-        liveList.setSelected(doc._id);
-
-        var refreshing = doc && $scope.selected && $scope.selected.id === doc._id;
-        $scope.selected = doc;
-        setTitle(doc);
-        $scope.setActionBar({
-          _id: doc._id,
-          verified: doc.verified,
-          type: doc.content_type,
-          sendTo: doc
-        });
+      $scope.setSelected = function(doc) {
+        var refreshing = true;
+        var displayFields = getDisplayFields(doc);
+        if ($scope.selectMode) {
+          var existing = _.findWhere($scope.selected, { _id: doc._id });
```

Comment:
`selected` is like `[{ report: { _id: 'aaa' }, expanded: false }, { report: { _id: 'bbb' }, expanded: false }]`
I don't think `_.findWhere` looks deeper than first level of attributes.
So I don't think `existing` is ever non-null!